api/src/utils/validate-query.ts

```diff
@@ -11,7 +11,10 @@ const querySchema = Joi.object({
 	group: Joi.array().items(Joi.string()),
 	sort: Joi.array().items(Joi.string()),
 	filter: Joi.object({}).unknown(),
-	limit: Joi.number().integer().min(-1),
+	limit:
+		'MAX_QUERY_LIMIT' in env && env.MAX_QUERY_LIMIT !== -1
+			? Joi.number().integer().min(-1).max(env.MAX_QUERY_LIMIT) // min should be 0
```

Comment:
I might have lost some context, but just to double check, is the comment "min should be 0" still accurate here?

Reply:
I think that comment is outdated as long as `limit=-1` still functions as the max when used in a filter.